### PR TITLE
Add Block 3 to main

### PR DESCRIPTION
**Pull request by karinorman**
This is the working pipeline for getting sample points from environmental layers.

### CODE REVIEW

**Comment by gottacatchenall** [comment]
might be worth doing #39 here as it is only a one-line change

**Comment by karinorman** [comment]
@gottacatchenall good call! done

**Comment by jmlord** [comment]
I tried running it and I got this :

```
ERROR: LoadError: ArgumentError: Package SimpleSDMLayers not found in current path.
- Run `import Pkg; Pkg.add("SimpleSDMLayers")` to install the SimpleSDMLayers package.
Stacktrace:
 [1] macro expansion
   @ ./loading.jl:1163 [inlined]
 [2] macro expansion
   @ ./lock.jl:223 [inlined]
 [3] require(into::Module, mod::Symbol)
   @ Base ./loading.jl:1144
in expression starting at /scripts/pointSelection/combineLayers.jl:2
```
I have build runner-julia locally to have Michhael's changes.

File: scripts/pointSelection/combineLayers.jl [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+W = Matrix(layermat[keep_rows, 2:end])
+
+#priority = SimpleSDMPredictor(squish(layers, convert(Array{Float64}, transpose(W)), targetbalance))
+priority = SimpleSDMPredictor(squish(layers, W, targetbalance))
+
+priority_path = joinpath(outputFilepath, "priority_map.tiff")
+###################
+print(priority_path)
+# write out the priority map
+geotiff(priority_path, priority)
```
Comment (jmlord) [inline scripts/pointSelection/combineLayers.jl:45]: In R, they have to specify parameters like COG and COMPRESS=DEFLATE to make sure that the file is a cloud optimized geotiff and not a regular geotiff, and that it isn't huge. It it necessary with this api?

Comment (gottacatchenall) [reply scripts/pointSelection/combineLayers.jl:45]: this is a super easy fix we discussed but did not implement, we can merge this soon, see https://github.com/PoisotLab/SpeciesDistributionToolkit.jl/issues/160

**Comment by gottacatchenall** [comment]
> I tried running it and I got this :
> 
> ```
> ERROR: LoadError: ArgumentError: Package SimpleSDMLayers not found in current path.
> - Run `import Pkg; Pkg.add("SimpleSDMLayers")` to install the SimpleSDMLayers package.
> Stacktrace:
>  [1] macro expansion
>    @ ./loading.jl:1163 [inlined]
>  [2] macro expansion
>    @ ./lock.jl:223 [inlined]
>  [3] require(into::Module, mod::Symbol)
>    @ Base ./loading.jl:1144
> in expression starting at /scripts/pointSelection/combineLayers.jl:2
> ```
> 
> I have build runner-julia locally to have Michael's changes.

can be fixed by replacing this line https://github.com/GEO-BON/biab-2.0/blob/27cfe8a3ac0a4e20481453ba3c24fab758fd9bd9/scripts/pointSelection/combineLayers.jl#L2 with `using SpeciesDistributionTooklit`

File: scripts/pointSelection/combineLayers.jl (hunk 1 above)
Comment (gottacatchenall) [reply scripts/pointSelection/combineLayers.jl:45]: currently SDT.jl uses `COMPRESS=LZW` as a default, but this could possibly also be a kwarg

**Comment by jmlord** [comment]
Hi again,

So after fixing the imports, I am still unable to run with default parameters.
```
[ Info: Loading BONs.jl support for SimpleSDMLayers.jl ...
Any["layer, current, change\nhttps://object-arbutus.cloud.computecanada.ca/bq-io/io/qc_pilot_env/backward_velocity.tif, 0.2, 0.8\nhttps://object-arbutus.cloud.computecanada.ca/bq-io/io/qc_pilot_env/backward_velocity.tif, 0.8, 0.2\n"]
2×3 DataFrame
 Row │ layer                               current   change
     │ String                             Float64   Float64
─────┼──────────────────────────────────────────────────────
   1 │ https://object-arbutus.cloud.com…       0.2      0.8
   2 │ https://object-arbutus.cloud.com…       0.8      0.2
/output/pointSelection/combineLayers/c1a0f833065e9eafbdfc9479364e1511/data/priority_map.tiff
ERROR: LoadError: UndefVarError: geotiff not defined
Stacktrace:
 [1] top-level scope
   @ /scripts/pointSelection/combineLayers.jl:45
in expression starting at /scripts/pointSelection/combineLayers.jl:45
Error: output.json file not found
```
I checked and both tiffs exist. Does this really run on your side?

**Comment by gottacatchenall** [comment]
`geotiff` is deprecated for reading tifs in the new package, I didn't run it last friday, I can try and run it later today and fix any final bugs

**Comment by gottacatchenall** [comment]
@jmlord this runs and outputs the coords as GeoJSON, but that can change to support #41

**Comment by jmlord** [comment]
> @jmlord this runs and outputs the coords as GeoJSON, but that can change to support #41

We can test it, but AFAIK the "dot for occurences", once merged, will apply to all markers, including the ones from a GeoJSON file.

**Comment by jmlord** [comment]
I had to do some changes to julia code to see the result in the UI.
However, the projection is not specified, and obviously it is not what leaflet expects:
image


I tried to add the CRS like this, but I don't know what to put as a projection. It is not a parameter to these steps... probably it should be?
``` julia
json_string = ("{
    \"type\": \"MultiPoint\",
    \"coordinates\": [
        $points_string_join
    ],
    \"crs\": {
        \"type\": \"name\",
        \"properties\": {
            \"name\": \"EPSG:3857\"
        }
    }
}")
```

**Comment by jmlord** [comment]
Please validate the results, not only that it runs.

The priority map makes the titiler server crash, sent it to @glaroc for validation and apparently in QGIS it looks like this:

image


Also, the same tiff is used twice as an input param. Maybe we should at least use two different tiffs so we could play with the parameters,

**Comment by gottacatchenall** [comment]
atm i'm not able to run anything on this branch, it aborts on the pipeline AssignId step with 

```
Creates an object carrying a layer and its identifier. To be used in conjunction with PullLayersById, inside a pipeline. (Cannot be ran as a single script.)

Skipped due to previous failure
```

What changed? @jmlord

**Comment by jmlord** [comment]
> atm i'm not able to run anything on this branch, it aborts on the pipeline AssignId step with
> 
> ```
> Creates an object carrying a layer and its identifier. To be used in conjunction with PullLayersById, inside a pipeline. (Cannot be ran as a single script.)
> 
> Skipped due to previous failure
> ```
> 
> What changed? @jmlord
Normally, when you see this, there is a yellow exclamation mark besides the ones that were skipped, and an error sign, in red, next to the one that caused the error. This one should have a more relevant error message!

Also, for this branch to work make sure you build the Julia environment. For that you need to change these comments:
``` yml
  runner-julia:
    container_name: biab-runner-julia
    # image: geobon/bon-in-a-box:runner-julia
    build:
      context: ./runners
      dockerfile: julia-dockerfile
```

**Pull request closed without merging**